packages/shared/src/sharing/generate-invite-code.test.ts

```diff
@@ -0,0 +1,44 @@
+import { describe, expect, it } from 'vitest';
+import { generateInviteCode } from './generate-invite-code';
+
+describe('generateInviteCode', () => {
+  it('should generate a code with default length of 8', () => {
+    const code = generateInviteCode();
+    expect(code).toHaveLength(8);
+  });
+
+  it('should generate a code with custom length', () => {
+    const code = generateInviteCode(12);
+    expect(code).toHaveLength(12);
+  });
+
+  it('should only contain allowed characters', () => {
+    const code = generateInviteCode();
+    const allowedChars = '123456789ABCDEFGHIJKLMNPQRSTUVWXYZ';
+
+    for (const char of code) {
+      expect(allowedChars).toContain(char);
+    }
+  });
+
+  it('should not contain potentially confusing characters (0, O)', () => {
+    const code = generateInviteCode();
+    const confusingChars = '0O';
+
+    for (const char of code) {
+      expect(confusingChars).not.toContain(char);
+    }
```

Comment:
Inverting the `for` would lead to better error messages.

```suggestion
    for (const char of confusingChars) {
      expect(code).not.toContain(char);
    }
```